src/binding/petsc4py/src/petsc4py/PETSc/Viewer.pyx

```diff
@@ -82,9 +86,33 @@ class DrawSize(object):
 # --------------------------------------------------------------------
 
 cdef class Viewer(Object):
+    """Viewer object.
+
+    Viewer is described in the `PETSc manual <petsc:sec_viewers>`.
+Viewers can be called as functions where the argument specified is the PETSc object to be viewed. See the example below.
+
+Examples
+--------------
+>>> from petsc4py import PETSc
+>>> u = PETSc.Vec().createWithArray([1,2])
+>>> v = PETSc.Viewer()
+>>> v(u)
+Vec Object: 1 MPI process
+  type: seq
+1.
+2.
+    See Also
+    --------
+    petsc.PetscViewer
+
+    """
 
     Type   = ViewerType
     Format = ViewerFormat
+    FileMode = FileMode
```

Comment:
If things are working, then you probably do not need this two new assignments in here.

Reply:
The problem (at least what I have understood) is that with the new documentation, we will have <img width="274" alt="Screen Shot 2023-04-03 at 9 08 41 PM" src="https://user-images.githubusercontent.com/5121281/229591836-bc9d56e2-8d54-4ccc-ac80-bb3959e06a45.png">. But then in code, you can access those enumerations via `Viewer.Mode.blah`, instead of `Viewer.FileMode.blah` as the documentation would suggest. Or am I wrong?

Reply:
But if you have `FileMode` at the top-level namespace, then writing `mode: FileMode | str` in function signature should resolve to the global `FileMode` outside of the `Viewer` class.

Let me clone your branch and take a final decision on this.